### PR TITLE
[KernelGen][Nvidia] Move _log_softmax_backward_data to ops

### PR DESCRIPTION
**Pull request by YufanPeter**
### Summary
Remove the duplicate experimental `_log_softmax_backward_data` and optimize the backward kernel. No registration changes are needed since `_log_softmax_backward_data` was already registered to `log_softmax_backward` in ops. The existing kernel's speedup was low, so this PR optimizes the kernel implementation based on the experimental version and removes the duplicate experimental copy.
### Testing
- Parametrized correctness tests for `_log_softmax_backward_data` across multiple shapes and all supported float dtypes **(18 Passed)**
- Validated against the PyTorch reference `torch.ops.aten._log_softmax_backward_data`
- Tested on: NVIDIA H20

### Performance
Tested on NVIDIA H20 using:
```bash
pytest -s benchmark/test_log_softmax.py --level core -k backward
```

#### _log_softmax_backward_data

**float16**
| shape | Torch latency | Gems latency | speedup |
|---|---|---|---|
| [64, 64] | 0.006 | 0.005 | 1.044x |
| [256, 256] | 0.006 | 0.006 | 1.117x |
| [1024, 1024] | 0.011 | 0.007 | 1.567x |
| [4096, 4096] | 0.039 | 0.035 | 1.105x |
| [1024, 65536] | 0.153 | 0.146 | 1.048x |

**float32**
| shape | Torch latency | Gems latency | speedup |
|---|---|---|---|
| [64, 64] | 0.005 | 0.006 | 0.977x |
| [256, 256] | 0.006 | 0.006 | 0.995x |
| [1024, 1024] | 0.011 | 0.008 | 1.291x |
| [4096, 4096] | 0.067 | 0.058 | 1.165x |
| [1024, 65536] | 0.359 | 0.347 | 1.035x |

**bfloat16**
| shape | Torch latency | Gems latency | speedup |
|---|---|---|---|
| [64, 64] | 0.005 | 0.005 | 0.991x |
| [256, 256] | 0.006 | 0.006 | 1.101x |
| [1024, 1024] | 0.011 | 0.007 | 1.571x |
| [4096, 4096] | 0.039 | 0.035 | 1.110x |
| [1024, 65536] | 0.154 | 0.154 | 1.000x |

#### _log_softmax_backward_data.out

**float16**
| shape | Torch latency | Gems latency | speedup |
|---|---|---|---|
| [64, 64] | 0.005 | 0.005 | 1.043x |
| [256, 256] | 0.006 | 0.006 | 1.123x |
| [1024, 1024] | 0.011 | 0.007 | 1.515x |
| [4096, 4096] | 0.038 | 0.037 | 1.043x |
| [1024, 65536] | 0.158 | 0.144 | 1.102x |

**float32**
| shape | Torch latency | Gems latency | speedup |
|---|---|---|---|
| [64, 64] | 0.005 | 0.005 | 1.012x |
| [256, 256] | 0.006 | 0.006 | 1.027x |
| [1024, 1024] | 0.011 | 0.009 | 1.197x |
| [4096, 4096] | 0.067 | 0.058 | 1.165x |
| [1024, 65536] | 0.383 | 0.352 | 1.085x |

**bfloat16**
| shape | Torch latency | Gems latency | speedup |
|---|---|---|---|
| [64, 64] | 0.006 | 0.006 | 1.023x |
| [256, 256] | 0.006 | 0.005 | 1.117x |
| [1024, 1024] | 0.011 | 0.008 | 1.417x |
| [4096, 4096] | 0.039 | 0.037 | 1.067x |
| [1024, 65536] | 0.154 | 0.144 | 1.068x |

#### Average
Combined average speedup: 1.137x

### Files changed
`src/flag_gems/ops/log_softmax.py`: Split backward kernel into inner / non-inner paths (reuses softmax tune configs read-only), fix tensor mask `and`→`&`
`src/flag_gems/experimental_ops/_log_softmax_backward_data.py`: Removed
`experimental_tests/unit/_log_softmax_backward_data_test.py`: Removed
`experimental_tests/performance/_log_softmax_backward_data_test.py`: Removed

### CODE REVIEW

**Comment by bin913** [comment]
Please add performance and correctness tests for this modification.

**Comment by bin913** [comment]
@YufanPeter Please add benchmark and accuracy tests
